bigtable-hbase-1.x-parent/bigtable-hbase-1.x-benchmarks/src/main/java/com/google/cloud/bigtable/hbase1_x/BigtableBenchmark.java

```diff
@@ -0,0 +1,211 @@
+  }
+
+  @Benchmark
+  public void pointRead(Blackhole blackhole) throws IOException {
+    Table table = connection.getTable(jobConfig.tableName);
+    String rowPrefix = getRowPrefix(jobConfig.cellsPerRow, jobConfig.cellSize);
+    Result result = table.get(new Get(Bytes.toBytes(rowPrefix + String.format(SUFFIX_FMT, 1))));
+
+    for (Cell cell : result.listCells()) {
+      blackhole.consume(cell);
+    }
+  }
+
+  @Benchmark
+  public void bulkScans(Blackhole blackhole) throws IOException {
+    String rowPrefix = getRowPrefix(jobConfig.cellsPerRow, jobConfig.cellSize);
+
+    Scan scan = new Scan().setMaxVersions(1).setRowPrefixFilter(Bytes.toBytes(rowPrefix));
+    ResultScanner resScanner = connection.getTable(jobConfig.tableName).getScanner(scan);
+
+    for (Result result = resScanner.next(); result != null; result = resScanner.next()) {
+
+      for (Cell cell : result.rawCells()) {
+        blackhole.consume(cell);
+      }
+    }
+  }
+
+  // TODO(rahulkql): yet to be fully tested.
+  @Benchmark
+  public void pointWrite() throws IOException {
+    Table table = connection.getTable(TableName.valueOf(BENCHMARK_WRITE_TABLE));
+
+    String rowPrefix = getRowPrefix(jobConfig.cellsPerRow, jobConfig.cellSize);
+    Put put = new Put(Bytes.toBytes(rowPrefix + String.format(SUFFIX_FMT, 100)));
+    long currentTimestamp = System.currentTimeMillis();
+
+    for (int cellInd = 0; cellInd < jobConfig.cellsPerRow; cellInd++) {
+      put.addColumn(
+          COL_FAMILY,
+          Bytes.toBytes("qualifier-" + String.format(SUFFIX_FMT, cellInd)),
+          currentTimestamp,
+          getRandomBytes(jobConfig.cellSize));
+    }
+
+    table.put(put);
+  }
+
+  // TODO(rahulkql): yet to be fully tested.
+  @Benchmark
+  public void bulkWrite() throws IOException {
+    try (BufferedMutator bufferedMutator =
+        connection.getBufferedMutator(TableName.valueOf(BENCHMARK_WRITE_TABLE))) {
+
+      String rowPrefix = getRowPrefix(jobConfig.cellsPerRow, jobConfig.cellSize);
+      byte[] cellValue = getRandomBytes(jobConfig.cellSize);
+      ImmutableList.Builder<Put> putBuilder = ImmutableList.builderWithExpectedSize(ROWS_PER_BATCH);
+      long currentTimestamp = System.currentTimeMillis();
+
+      for (int rowInd = 0, sendBatchInd = 0;
+          rowInd < jobConfig.totalRows;
+          rowInd++, sendBatchInd++) {
+
+        // zero padded row-key
+        Put put = new Put(Bytes.toBytes(rowPrefix + String.format(SUFFIX_FMT, rowInd)));
+
+        for (int cellInd = 0; cellInd < jobConfig.cellsPerRow; cellInd++) {
+          put.addColumn(
+              COL_FAMILY,
+              Bytes.toBytes("qualifier-" + String.format(SUFFIX_FMT, cellInd)),
+              currentTimestamp,
+              cellValue);
+        }
+        putBuilder.add(put);
+
+        // Sending a lot of rows in a single batch randomly OOM exception
+        if (sendBatchInd % ROWS_PER_BATCH == 0 || rowInd == (jobConfig.totalRows - 1)) {
```

Comment:
This benchmark should only send a single batch. Let JMH call this multiple times instead